File: Practice/src/coding_basic_training/ReverseStringMultipleTimes.java
package coding_basic_training;

public class ReverseStringMultipleTimes {
//	문자열 여러 번 뒤집기
	public String solution(String my_string, int[][] queries) {
		char[] c = my_string.toCharArray();
		for (int i = 0; i < queries.length; i++) {
			int x = queries[i][0];
			int y = queries[i][1];
			for (int j = x; j < x + (y - x + 1) / 2; j++) {
				char temp = c[j];
				c[j] = c[y - (j - x)];
				c[y - (j - x)] = temp;
			}
		}
		return new String(c);
	}
}

File: Practice/src/coding_basic_training/ReverseStringMultipleTimes.js
function solution(my_string, queries) {
    let answer = my_string.split("");
    for (let [x, y] of queries) {
        for (let i = x; i < x + (y - x + 1) / 2; i++) {
            let temp = answer[i];
            answer[i] = answer[y - i + x];
            answer[y - i + x] = temp;
        }
    }
    return answer.join("");
}
